src/views/blocks/event-datetime.php

```diff
@@ -80,34 +81,36 @@
 			<span class="tribe-events-schedule__all-day"><?php echo __( 'All day', 'the-events-calendar' ); ?></span>
 		<?php endif; ?>
 
-		<?php if ( ! $is_all_day || ! $is_same_day ) : ?>
-			<span class="tribe-events-schedule__separator tribe-events-schedule__separator--time">
-				<?php echo $separator_time; ?>
-			</span>
-		<?php endif; ?>
-
-		<?php if ( ! $is_same_day ) : ?>
-			<span class="tribe-events-schedule__date tribe-events-schedule__date--end">
-				<?php echo $formatted_end_date; ?>
-			</span>
+		<?php if ( ! $is_same_start_end ) : ?>
+			<?php if ( ! $is_all_day || ! $is_same_day ) : ?>
+				<span class="tribe-events-schedule__separator tribe-events-schedule__separator--time">
+					<?php echo $separator_time; ?>
+				</span>
+			<?php endif; ?>
 
-			<?php if ( ! $is_all_day ) : ?>
-				<span class="tribe-events-schedule__separator tribe-events-schedule__separator--date">
-					<?php echo $separator_date; ?>
+			<?php if ( ! $is_same_day ) : ?>
+				<span class="tribe-events-schedule__date tribe-events-schedule__date--end">
+					<?php echo $formatted_end_date; ?>
 				</span>
+
+				<?php if ( ! $is_all_day ) : ?>
+					<span class="tribe-events-schedule__separator tribe-events-schedule__separator--date">
+						<?php echo $separator_date; ?>
```

Comment:
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$separator_date'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$separator_date'.

<sub>hash: 3ecf734af43754818ab5e188efadc04d</sub>